src/controllers/sessionController.ts

```diff
@@ -15,20 +15,63 @@ import { PublicSession } from "../../generated/backend/PublicSession";
 import { SessionsList } from "../../generated/backend/SessionsList";
 
 import { isLeft } from "fp-ts/lib/Either";
+import TokenService from "src/services/tokenService";
+import { MyPortalToken } from "src/types/token";
 import RedisSessionStorage from "../services/redisSessionStorage";
-import { withUserFromRequest } from "../types/user";
+import { UserV2, withUserFromRequest } from "../types/user";
+import { SESSION_TOKEN_LENGTH_BYTES } from "./authenticationController";
+
+import { log } from "../utils/logger";
 
 export default class SessionController {
-  constructor(private readonly sessionStorage: RedisSessionStorage) {}
+  constructor(
+    private readonly sessionStorage: RedisSessionStorage,
+    private readonly tokenService: TokenService
+  ) {}
   public readonly getSessionState = (
     req: express.Request
-  ): Promise<IResponseErrorValidation | IResponseSuccessJson<PublicSession>> =>
-    withUserFromRequest(req, async user =>
-      ResponseSuccessJson({
-        spidLevel: user.spid_level,
-        walletToken: user.wallet_token
-      })
-    );
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseSuccessJson<PublicSession>
+  > =>
+    withUserFromRequest(req, async user => {
+      if (UserV2.is(user)) {
```

Comment:
Can we track with applicationInsights how many UserV1 hits we have ? This value will converge to zero so we can remove the mapping in a near future

Reply:
All user sessions have a lifetime of 30 days, so we can safely remove the mapping after that time. Do we still need an event for applicationInsights?

Reply:
nope, that's right!